pkg/server/v3/delta_test.go

```diff
@@ -379,3 +379,82 @@ func TestDeltaAggregatedHandlers(t *testing.T) {
 		}
 	}
 }
+
+func TestDeltaAggregateRequestType(t *testing.T) {
+	config := makeMockConfigWatcher()
+	s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+	resp := makeMockDeltaStream(t)
+	resp.recv <- &discovery.DeltaDiscoveryRequest{Node: node}
+	if err := s.DeltaAggregatedResources(resp); err == nil {
+		t.Error("DeltaAggregatedResources() => got nil, want an error")
+	}
+}
+
+func TestDeltaCancellations(t *testing.T) {
+	config := makeMockConfigWatcher()
+	resp := makeMockDeltaStream(t)
+	for _, typ := range testTypes {
+		resp.recv <- &discovery.DeltaDiscoveryRequest{
+			Node:    node,
+			TypeUrl: typ,
+		}
+	}
+	close(resp.recv)
+	s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+	if err := s.DeltaAggregatedResources(resp); err != nil {
+		t.Errorf("DeltaAggregatedResources() => got %v, want no error", err)
+	}
+	if config.watches != 0 {
+		t.Errorf("Expect all watches cancelled, got %q", config.watches)
+	}
+}
+
+func TestDeltaOpaqueRequestsChannelMuxing(t *testing.T) {
+	config := makeMockConfigWatcher()
+	resp := makeMockDeltaStream(t)
+	for i := 0; i < 10; i++ {
+		resp.recv <- &discovery.DeltaDiscoveryRequest{
+			Node:    node,
+			TypeUrl: fmt.Sprintf("%s%d", opaqueType, i%2),
+			// each subsequent request is assumed to supercede the previous request
```

Comment:
supercede seems like the wrong term here since each request is for a different type?

Reply:
Maybe? I think they're meaning to say each request takes places of the previous one. Could also change that to `succeeds`?

Reply:
If we don't know what it means maybe just remove it :)

Reply:
Ha works for me